src/runtime.py

```diff
@@ -1,31 +1,35 @@
-from src.objects.agent import Agent
+from src.actions.next_scene import NextScene
 from src.common.enums import Direction
-from src.common.point import Point
 from src.actions.move import Move
+from src.common.exec_interrupt import ExecInterrupt
+from src.scene import Scene
 
 
 class Runtime:
-    MAX_SPINS: int = 8
+    '''
+    Supplies basic execution of users code across various levels.
+    '''
 
     def __init__(self, agents_code: str) -> None:
         self.agents_code: str = agents_code
+        self.scene: Scene = None
+        self.scenes = []
+        self.next_scene = NextScene(self)
```

Comment:
Also I don't like passing runtime to scene
If it requires passing self, it means it should be done in runtime itself

Reply:
But it's an `Action` for changing the state of the runtime, so it cannot operate without the reference.